.github/workflows/tests.yaml

```diff
@@ -74,33 +81,43 @@ jobs:
         if: steps.filter.outputs.has_code == 'true'
         with:
           # Match the min version listed in docs/project/contribution_tools.md
-          python-version: '3.9'
+          # or the oldest version available on the OS.
+          python-version: ${{ matrix.os == 'macos-14' && '3.11' || '3.9' }}
```

Comment:
Had you considered updating the min version? I don't think we've changed for a couple years now, maybe updating it to 3.11 everywhere would be fine?

Reply:
Happy to try that, but seemed like maybe better to do separately to minimize how much changes with this PR?

Reply:
SGTM